azerothcore/azerothcore-wotlk
# fix(Core/Pets): Master Demonologist should not be removed on pet resu…

…mmon.

Fixes #8994



## Issues Addressed:

- Closes #8994
- Closes https://github.com/chromiecraft/chromiecraft/issues/2267

## Tests Performed:

- Tested ingame.

## How to Test the Changes:

1. Imp Re-summon of current pet: disable of talent aura and buff icon
2. Succubus 1st summon: enable of talent aura and buff icon, but stats doesn't show shadow crit chance
3. Succubus Re-summon of current pet: disable of talent aura and buff icon, still nothing in stats
4. Switching to other pet restore to normal state
5. Sometimes after dismiss pet you can see presenting + crit, but it's just visual bug


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
